### PR TITLE
Add implementation for `apply_weights()` function

### PR DESCRIPTION
**Pull request by andersy005**
This PR implements `apply_weights()` function. We are relying on `xarray.dot()` which takes care of:

- broadcasting
- chunks alignment when mapping DataArrays containing dask arrays. 

## TODO 

- [ ] Add `apply_weights()` function
- [ ] Add `__call__()` method
- [ ] Add helper functions for sanitizing input data and finalizing output data
- [ ] Update tests
- [ ] Update docstrings
- [ ] Update Demo notebook

[Demo notebook](https://nbviewer.jupyter.org/github/andersy005/xtimeutil/blob/average/docs/source/notebooks/tutorial.ipynb)

@dcherian, when you get a chance, I could use your feedback on this :)

### CODE REVIEW

File: xtimeutil/remapper.py [hunk 1]
```diff
@@ -199,13 +208,45 @@ def get_outgoing_time_axis_info(self):
...
+                )
+            else:
+                outdata = _mean(self.weights, indata, self.incoming['time_coord_name'])
+            outdata = _finalize_output_data(
+                outdata, self.incoming['time_coord_name'], self.outgoing, self.incoming
+            )
+            return outdata
+
+        else:
+            raise NotImplementedError
```
Comment (kmpaul) [inline xtimeutil/remapper.py:233]: Might be good to tell people *why* they are getting a `NotImplementedError` (namely, what isn't implemented?):

```suggestion
            raise NotImplementedError('Currently only works on xarray DataArrays')
```

File: xtimeutil/remapper.py [hunk 2]
```diff
@@ -199,13 +208,45 @@ def get_outgoing_time_axis_info(self):
         x.update(attrs)
         return x
 
+    def mean(self, data):
+        if isinstance(data, xr.DataArray):
```
Comment (kmpaul) [inline xtimeutil/remapper.py:212]: You could also make this a "check-raise" statement with:

```python
if not isinstance(data, xr.DataArray):
    raise NotImplementedError('mean currently only works with xarray DataArrays')
```

Then the remaining code in the function need not be indented, but this is not a necessary change.

**Comment by kmpaul** [comment]
@andersy005 What is the `mean` function actually doing?  It looks to me (with the exception of some corner case handling) that `mean` actually computes the data variable on the remapped coordinate...?  In other words, it returns the matrix multiplication of the weights matrix with the input data, correct?  Why is this `mean`?  Shouldn't it be something like `remap`?  Or, since the object is called a `Remapper`, couldn't it just be the `__call__` function?

**Comment by kmpaul** [comment]
I would also like to see more examples in the notebook.  It seems hard to understand what it is actually *doing* from the notebook.  (Or maybe you just need to highlight what the reader should be looking at.)

**Comment by andersy005** [comment]
> Some minor code suggestions, but I don't quite get what you are doing.

:( I am going to turn this PR into a draft for now, and will spend some time improving it, and will ping you when it's ready for another round of review. Thank you for the suggestions! 


> It seems hard to understand what it is actually doing from the notebook. (Or maybe you just need to highlight what the reader should be looking at.)

You are right. Looking at the notebook a bit, I think the notebook itself is half-baked. I will update it with more details/examples.

**Comment by andersy005** [comment]
Closing this PR for now. It will be superseded by another PR in the future.

**Pull request closed without merging**